### ISSUE TITLE
Dependency on Sinon 11 causes type errors when compiling TypeScript?

### ISSUE THREAD

**Issue by nickmason** [m-radzikowski/aws-sdk-client-mock#130]
**Checklist**

- [x] I have read [Caveats](https://github.com/m-radzikowski/aws-sdk-client-mock#caveats) documentation and didn't find a solution for this problem there.

**Bug description**

`aws-sdk-client-mock` has a dependency on `^sinon@11.1.1`. This is an older version of Sinon that included its own types, which have since been removed in subsequent versions and replaced with the `@types/sinon` package, but it seems that built-in types take precedence over `@types/`.

This appears to be causing the problem mentioned in [this discussion](https://github.com/DefinitelyTyped/DefinitelyTyped/discussions/61374) over at the DefinitelyTyped repo, when you try to build a project with that version of Sinon anywhere in your dependency tree, you get this:

```
$ ./node_modules/.bin/tsc
node_modules/@types/sinon/index.d.ts:1447:53 - error TS2694: Namespace '"/Users/nmason/Source/sinon-aws-client-mock/node_modules/@sinonjs/fake-timers/types/fake-timers-src"' has no exported member 'FakeTimerInstallOpts'.

1447         useFakeTimers: boolean | Partial<FakeTimers.FakeTimerInstallOpts>;
                                                         ~~~~~~~~~~~~~~~~~~~~

node_modules/@types/sinon/index.d.ts:1569:67 - error TS2694: Namespace '"/Users/nmason/Source/sinon-aws-client-mock/node_modules/@sinonjs/fake-timers/types/fake-timers-src"' has no exported member 'FakeTimerInstallOpts'.

1569         useFakeTimers(config?: number | Date | Partial<FakeTimers.FakeTimerInstallOpts>): SinonFakeTimers;
                                                                       ~~~~~~~~~~~~~~~~~~~~


Found 2 errors in the same file, starting at: node_modules/@types/sinon/index.d.ts:1447
```

**Reproduction**
Check out the [very minimal repository](https://github.com/nickmason/sinon-aws-client-mock), install dependencies, run `npm run build`.

Uninstall `aws-sdk-client-mock` and `tsc` happily builds the project without errors.

**Environment**

- Node version: 16.18.1
- Typescript version: 4.8.4
- AWS SDK v3 Client mock version: 2.0.0

**Comment by m-radzikowski** [comment]
Hey,
an immediate solution is to put `compilerOptions.skipLibCheck` to `true` in `tsconfig.json`.

The sinon v11 does not have TS types included, they are added as a separate dependency (`@types/sinon`). But since the mock instance is basically a Sinon stub, I had to put the types as a "regular" dependency, not a dev one, or TS would complain on lack of types.

Since the issue with the FakeTimers was introduced in `@types/sinon` v10.0.13, and it's the latest version, an alternative is to use the previous version (v10.0.12). If you add it by yourself to the package.json it will force using this version and will solve the problem as well.

I'm a bit reluctant to change the required `@types/sinon` version in the library. Right now it's `^10.0.10`. If I hardcode it to exactly `10.0.12`, I guess there will be people using sinon or other libs depending on it for whom it will cause version conflicts.

**Comment by nickmason** [comment]
Yeah that makes sense! I was thinking more along the lines of "How huge a pain in the butt would it to be bump Sinon as a whole to a version higher than 11.x so as to just sidestep this thing entirely". :) Current is 14.x apparently.

**Comment by m-radzikowski** [comment]
Hey, I was wrong about bumping the Sinon version - it actually bumped the`@sinonjs/fake-timers` and solved the issue 🙂

Thanks to the @samchungy for double-checking it and making a PR.

The fix is released in v2.0.1.
